src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,201 @@
+  };
+
+  using options = tmpl::list<BeamWidth, Material, IntegrationIntervals,
+                             AbsoluteTolerance, RelativeTolerance>;
+  static constexpr OptionString help{
+      "A semi-infinite mirror on which a laser introduces stress perpendicular "
+      "to the mirrors surface. The displacement then is the Hankel-Transform "
+      "of the general solution multiplied by the beam profile"};
+
+  HalfSpaceMirror() = default;
+  HalfSpaceMirror(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror& operator=(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror(HalfSpaceMirror&&) noexcept = default;
+  HalfSpaceMirror& operator=(HalfSpaceMirror&&) noexcept = default;
+  ~HalfSpaceMirror() noexcept = default;
+
+  HalfSpaceMirror(double beam_width,
+                  constitutive_relation_type constitutive_relation,
+                  size_t integration_intervals = 350,
+                  double absolute_tolerance = 1e-12,
+                  double relative_tolerance = 1e-10) noexcept;
+
+  const constitutive_relation_type& constitutive_relation() const noexcept {
+    return constitutive_relation_;
+  }
+
+  // @{
+  /// Retrieve variable at coordinates `x`
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Displacement<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Displacement<3>>;
+
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Strain<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Strain<3>>;
+
+  // FixedSource in Elasticity equates to external forces
+  // (see Elasticity::FirstOrderSystem)
+  static auto variables(
+      const tnsr::I<DataVector, 3>& x,
+      tmpl::list<::Tags::FixedSource<Tags::Displacement<3>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<::Tags::FixedSource<Tags::Displacement<3>>>;
+  // @}
+
+  /// Retrieve a collection of variables at coordinates `x`
+  template <typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataVector, 3>& x,
+                                         tmpl::list<Tags...> /*meta*/) const
+      noexcept {
+    static_assert(sizeof...(Tags) > 1, "An unsupported Tag was requested.");
+    for (size_t i = 0; i < get<2>(x).size(); i++) {
+      if (UNLIKELY(get<2>(x)[i] < 0)) {
+        ERROR(
+            "The HalfSpaceMirror solution is not defined for negative values "
+            "of z.");
+      }
+    }
+    return {tuples::get<Tags>(variables(x, tmpl::list<Tags>{}))...};
+  }
+
+  // clang-tidy: no pass by reference
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  friend bool operator==(const HalfSpaceMirror& lhs,
+                         const HalfSpaceMirror& rhs) noexcept;
+
+  double beam_width_{std::numeric_limits<double>::signaling_NaN()};
+  constitutive_relation_type constitutive_relation_{};
+  size_t integration_intervals_{};
+  double absolute_tolerance_{};
+  double relative_tolerance_{};
```

Comment:
default-construct the variables in lines 193--195 with `std::numeric_limits<size_t>::max()` or `std::numeric_limits<double>::signaling_NaN()` (like you do in line 191).